### PR TITLE
NMS-13168: Move Enlinkd models and related class from web-app module t…

### PR DESCRIPTION
**Pull request by jadyhou**
Move the Enlinkd related java code from web-app to web-api so that both web-app and web-rest can use it. This is a subtask of https://issues.opennms.org/browse/NMS-12941 . I want to do this first to make sure I didn't break the existing webapp function and then I'll add the restful API. 

### All Contributors

* [ ] Have you read and followed our [Contribution Guidelines](https://github.com/OpenNMS/opennms/blob/develop/CONTRIBUTING.md)?
* [ ] Have you [made an issue in the OpenNMS issue tracker](https://issues.opennms.org)?<br>If so, you should:
  1. update the title of this PR to be of the format: `${JIRA-ISSUE-NUMBER}: subject of pull request`
  2. update the JIRA link at the bottom of this comment to refer to the real issue number
  3. prefix your commit messages with the issue number, if possible
* [ ] Have you made a comment in that issue which points back to this PR?
* [ ] Have you updated the JIRA link at the bottom of this comment to link to your issue?
* [ ] If this is a new or updated feature, is there documentation for the new behavior?
* [ ] If this is new code, are there unit and/or integration tests?
* [ ] If this PR targets a `foundation-*` branch, does it avoid changing files in `$OPENNMS_HOME/etc/`?

### Pull Request Process

One or more reviewers should be assigned to each PR.

If you know that a particular person is subject matter expert in the area your PR affects, feel free to assign one or more reviewers when you create this PR, otherwise reviewers will be assigned for you.

If you have made additions or changes to the documentation, or if you _need_ documentation for these code changes, please make sure a technical writer has looked it over.

Once the reviewer(s) accept the PR and the branch passes continuous integration, the PR is eligible for merge.

At that time, if you have commit access (are an OpenNMS Group employee or a member of the Order of the Green Polo) you are welcome to merge the PR.
Otherwise, a reviewer can merge it for you.

Thanks for taking time to contribute!

### External References

* JIRA (Issue Tracker): https://issues.opennms.org/browse/NMS-13168

### CODE REVIEW

**Review by RangerRick — APPROVED** [review]
LOL, these were probably in the webapp from _before_ there was an `opennms-web-api`  :D

File: opennms-web-api/pom.xml [hunk 1]
```diff
@@ -85,6 +85,14 @@
...
       <type>pom</type>
     </dependency>
+    <dependency>
+      <groupId>org.opennms.features.enlinkd</groupId>
+      <artifactId>org.opennms.features.enlinkd.persistence.api</artifactId>
+    </dependency>
+    <dependency>
+      <groupId>org.opennms.features.enlinkd</groupId>
+      <artifactId>org.opennms.features.enlinkd.service.api</artifactId>
+    </dependency>
```
Comment (mattixtech) [inline opennms-web-api/pom.xml:95]: Can the corresponding deps be removed from the `opennms-webapp` pom now?

Comment (jadyhou) [reply opennms-web-api/pom.xml:95]: Good catch, Thanks!  Please see the update.

**Comment by jadyhou** [comment]
> LOL, these were probably in the webapp from _before_ there was an `opennms-web-api` :D

Yeah, that's quite normal. It seems not possible to put everything in place at once since there are so many moving parts.

**Review by mattixtech — APPROVED** [review]
🚀

**Pull request merged**